developer-docs-site/docs/tutorials/full-node/troubleshooting-fullnode.md

```diff
@@ -53,3 +53,15 @@ full_node_networks:
 [devnet_waypoint]: https://devnet.aptoslabs.com/waypoint.txt
 [aptos-labs/aptos-core]: https://github.com/aptos-labs/aptos-core.git
 [status dashboard]: https://status.devnet.aptos.dev
+
+## Incentivized Testnet Setup
+
+**Q: Generating only 1 key file instead of 3.**
+
+I am following [Run Validator Node Using Source Code](../validator-node/using-source-code.md) tutorial. One of my servers generates only 1 key file instead of 3, at step 7.
+
+**A:** Make sure you checkout the `testnet` branch as described in step 5.
+
+**Q: Do we need to run validator node and FullNode on two different machines, in order to get two different IPs?**
+
+**A:** You can provide same IP for both the validator node and the FullNode. Also, the FullNode IP is optional and you can leave it as blank. 
```

Comment:
per the note earlier, we should update documentation to suggest that individuals run nodes on well provisioned, isolated, and independent machines.